python/triton/compiler/compiler.py

```diff
@@ -278,8 +211,10 @@ def make_hash(fn, arch, **kwargs):
     "ttgir": mlir_arg_type_pattern,
     "ptx": ptx_arg_type_pattern,
 }
-
-ttgir_num_warps_pattern = r'"triton_gpu.num-warps"\s?=\s?(\d+)\s?:'
+if is_hip():
+    ttgir_num_warps_pattern = r'"triton_gpu_rocm.num-warps"\s?=\s?(\d+)\s?:'
+else:
+    ttgir_num_warps_pattern = r'"triton_gpu.num-warps"\s?=\s?(\d+)\s?:'
```

Comment:
Should something be added to allow backends to register a custom warp pattern instead of hard coding in either a binary of AMD or the default NVIDIA?

Reply:
the pattern is the same for both devices. the issues is that we are using a different gpu dialect. so maybe a way of registering gpu dialect names might be needed. I think this should be done in a subsquent pr to contain the scope of this. Let me know if you disagree.

Reply:
Due to the scope of the large scope of this pr and all the moving pieces, this will be postponed to a follow up pr.